src/components/themed/MiniButton.js

```diff
@@ -0,0 +1,98 @@
+// @flow
+
+import * as React from 'react'
+import { ActivityIndicator, Text, TouchableOpacity } from 'react-native'
+import { cacheStyles } from 'react-native-patina'
+
+import { useState } from '../../types/reactHooks.js'
+import { fixSides, mapSides, sidesToMargin } from '../../util/sides.js'
+import { type Theme, useTheme } from '../services/ThemeContext.js'
+
+type Props = {|
+  // Called when the user presses the button.
+  // If the callback returns a promise, the button will disable itself
+  // and show a spinner until the promise resolves.
+  onPress?: () => void | Promise<void>,
+
+  // Whether to center the button or stretch to fill the screen.
+  // Defaults to 'auto', letting the parent component be in charge:
+  alignSelf?: 'auto' | 'stretch' | 'center',
+
+  // True to dim the button & prevent interactions:
+  disabled?: boolean,
+
+  // If this is set, the component will insert a text node before the other children:
+  label?: string,
```

Comment:
This should not be optional, since the component doesn't take other children.

Reply:
Agreed. It should not be optional if there is not a `children` prop. If we later add `children` prop we would need to make this optional again.